crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -678,6 +687,51 @@ impl FileSystem for DiskFileSystem {
 
         let _lock = self.mutex_map.lock(full_path.clone()).await;
 
+        // Make sure that there is only one authorative writer at a time.
+        let conflict = {
+            let own_invalidator = get_invalidator();
+            let mut authorative_write_map = self.authorative_write_map.lock().unwrap();
+            let inner_map = authorative_write_map.entry(full_path.clone()).or_default();
+            let mut new_map = AutoMap::new();
```

Comment:
I don't see a path for ever inserting a value into `new_map`. Yes, we're inserting values from the `inner_map`, but that's just the previous value of `new_map`, so we could never have inserted something into the `inner_map` to begin with. There's also a path if there's a conflict, but the map can never start with a conflict, so we don't ever enter that code either.

I think we're missing a `new_map.insert(own_invalidator, …)` right after this line, which means we don't really need to track `has_own` at all (it'll always have own).

Reply:
yeah, that's missing...